vscode/webviews/chat/ChatMessageContent/ChatMessageContent.tsx

```diff
@@ -39,6 +39,47 @@ interface ChatMessageContentProps {
     className?: string
 }
 
+interface StreamingContent {
+    displayContent: string
+    thinkContent: string
+    hasThinkTag: boolean
+    isThinking: boolean
+}
+
+const extractThinkContent = (content: string): StreamingContent => {
+    const thinkRegex = /<think>([\s\S]*?)<\/think>/g
```

Comment:
```suggestion
    const thinkRegex = /^<think>([\s\S]*?)<\/think>/g
```
Do we want to make sure only the <think> tags at the beginning are being considered?  I know I wrote the original regex but haven't really tested it throughoutly 😆

Reply:
Also, do we want to consider "thinking" in additional to "think" for other provider? For example, Anthropic suggested using "<thinking>" tags for CoT https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/chain-of-thought#how-to-prompt-for-thinking